src/plugins/packages/index.js

```diff
@@ -35,38 +35,43 @@ function getPackage(name) {
 }
 
 function loadPackages() {
-	fs.readdir(Helper.getPackagesPath(), (err, packages) => {
-		if (err) {
-			return;
-		}
-		packages.forEach((packageName) => {
-			const packageFile = getModuleInfo(packageName);
-			if (!packageFile) {
-				return;
-			}
-			packageMap.set(packageName, packageFile);
-			if (packageFile.type === "theme") {
-				themes.addTheme(packageName, packageFile);
-			}
-
-			if (packageFile.onServerStart) {
-				packageFile.onServerStart(packageApis(packageName));
-			}
-		});
-	});
-}
+	const packageJson = path.join(Helper.getPackagesPath(), "package.json");
+	let packages;
 
-function getModuleInfo(packageName) {
-	let module;
 	try {
-		module = require(Helper.getPackageModulePath(packageName));
+		packages = Object.keys(require(packageJson).dependencies);
 	} catch (e) {
-		log.warn(`Specified package ${colors.yellow(packageName)} is not installed in packages directory`);
-		return;
-	}
-	if (!module.thelounge) {
-		log.warn(`Specified package ${colors.yellow(packageName)} doesn't have required information.`);
-		return;
+		packages = [];
 	}
-	return module.thelounge;
+
+	packages.forEach((packageName) => {
+		const errorMsg = `Package ${colors.bold(packageName)} could not be loaded`;
+		let packageInfo;
+		let packageFile;
+
+		try {
+			packageInfo = require(path.join(Helper.getPackageModulePath(packageName), "package.json"));
+			packageFile = require(Helper.getPackageModulePath(packageName));
+		} catch (e) {
+			setTimeout(() => log.warn(errorMsg));
+			return;
+		}
+
+		if (!packageInfo.thelounge) {
+			setTimeout(() => log.warn(errorMsg));
+			return;
+		}
+
+		packageMap.set(packageName, packageFile);
+
+		if (packageInfo.type === "theme") {
+			themes.addTheme(packageName, packageInfo);
+		}
+
+		if (packageFile.onServerStart) {
+			packageFile.onServerStart(packageApis(packageName));
+		}
+
+		setTimeout(() => log.info(`Package ${colors.bold(packageName)} loaded`));
```

Comment:
This `setTimeout` is very much subpar, but I don't want the logs to show first (because they're loaded before anything. In fact, I know how to do that in a much better way (and also address a rather nasty security concern I have right now), but in the meantime, I figured it would do. I'll clean up the tests as well (they need to use `done()` for now, not great).

Reply:
So wait, why are you doing this? You want them to not show before what?

Reply:
Oh wait, I see. You want them logged after the server is up. I'm not sure that's a great idea.

It makes it look like it turns the server on and then loads packages, rather than the other way around, which is misleading.

Also, all we need is 1 more async thing to happen before these and the logs will be messed up again. I say just log them in the order you get them.